scico/linop/abel.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2022 by SCICO Developers
+# All rights reserved. BSD 3-clause License.
+# This file is part of the SCICO package. Details of the copyright and
+# user license can be found in the 'LICENSE' file distributed with the
+# package.
+
+"""Abel transform LinearOperator wrapping the pyabel package.
+
+Abel transform LinearOperator wrapping the
+`pyabel <https://github.com/PyAbel/PyAbel>`_ package.
+"""
+
+import math
+from typing import Optional
+
+import numpy as np
+
+import jax
+import jax.numpy as jnp
+import jax.numpy.fft as jnfft
+
+import abel
+
+from scico.linop import LinearOperator
+from scico.typing import Array, JaxArray, Shape
+from scipy.linalg import solve_triangular
+
+
+class AbelProjector(LinearOperator):
+    r"""Abel transform projector based on `PyAbel <https://github.com/PyAbel/PyAbel>`_.
+
+    Perform Abel transform (parallel beam tomographic projection of
+    cylindrically symmetric objects) for a 2D image. The input 2D image
+    is assumed to be centered and left-right symmetric.
+    """
+
+    def __init__(self, img_shape: Shape):
+        """
+        Args:
+            img_shape: Shape of the input image.
+        """
+        self.proj_mat_quad = _pyabel_daun_get_proj_matrix(img_shape)
```

Comment:
Should this (`self.proj_mat_quad`) have type annotation?

Reply:
The `_pyabel_daun_get_proj_matrix` function has an output type annotation. Did you mean adding it in the body fo the `__init__` function?

Reply:
Yes. I do not know how strict are we supposed to be with this.

Reply:
For now the primary requirement is type annotations for functions and methods. Once we have type checking working (see #176) we can start worrying about typing annotations for attributes/variables.

Reply:
So, shall we postpone this to a later time when #176 is done?

Reply:
Yes.